charts/datadog/templates/_kubernetes_apiserver_config.yaml

```diff
@@ -1,13 +1,19 @@
 {{- define "kubernetes_apiserver-config" -}}
-{{- if .Values.datadog.collectEvents -}}
+{{- if or (.Values.datadog.collectEvents) (.Values.clusterAgent.kubernetesApiserverCheck.disableUseComponentStatus) -}}
```

Comment:
Since we have multiple conditions, I think this could be moved to the helper file similar to how we have [need-cluster-agent-confd](https://github.com/DataDog/helm-charts/blob/4a45732dcf5131d40bcd4b709370d5d0cbfc9f3b/charts/datadog/templates/_helpers.tpl#L701) for the cluster agent confd. The helper could then be used in the change you have for `charts/datadog/templates/cluster-agent-deployment.yaml`

Reply:
@khewonc I see, created `need-kuberetes-apiserver-check-config` in helper and updated both.